src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,361 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import java.util.List;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int STANDARD_DB_SET_COUNTER = 3;
+    private static final String EXPECTED_MESSAGE_ADDED_TO_DB_RETURN_USER =
+            "Expected, that user will be return after added to the DB with correct data";
+    private static final String EXPECTED_MESSAGE_ADDED_TO_DB =
+            "Expected, that user will be added to the DB with correct data";
+    private static final String EXPECTED_MESSAGE_NULL =
+            "Expected, that existing user can't be added to the DB, must return null";
+    private static final String EXPECTED_MESSAGE_NULL_USER =
+            "Expected, that null pointer instead User have to call the RegistrationException";
+    private static final String EXPECTED_MESSAGE_NULL_LOGIN_USER =
+            "Expected, that user with null login call the RegistrationException";
+    private static final String EXPECTED_MESSAGE_INVALID_LOGIN_USER =
+            "Expected, that user with invalid login call yhe RegistrationException";
+    private static final String EXPECTED_MESSAGE_NULL_AGE_USER =
+            "Expected, that user with null age call yhe RegistrationException";
+    private static final String EXPECTED_MESSAGE_INVALID_AGE_USER =
+            "Expected, that user with invalid login call yhe RegistrationException";
+    private static final String EXPECTED_MESSAGE_NULL_PASSWORD_USER =
+            "Expected, that user with null password call yhe RegistrationException";
+    private static final String EXPECTED_MESSAGE_INVALID_PASSWORD_USER =
+            "Expected, that user with invalid password call yhe RegistrationException";
+    private static final int ADDED_FIRST_CUSTOM_USER = 3;
+    private static final int ADDED_SECOND_CUSTOM_USER = 4;
+    private static final int ADDED_THIRD_CUSTOM_USER = 5;
+
+    private static User userBob;
+    private static User userAnn;
+    private static User userRoma;
+    private RegistrationService registrationService;
+
+    @BeforeAll
+    static void setDataBase() {
+        userBob = new User("user_bob", "qwerty1234", 34);
+        userAnn = new User("user_ann", "12345678", 18);
+        userRoma = new User("user_roma", "bobocode", 22);
+        Storage.people.addAll(List.of(userAnn, userBob, userRoma));
```

Comment:
In general, I understand and agree with the logic of the tests, but I have one question: is it necessary to add as many as 3 users to the database for clarity? I would use one, and thus reduce the amount of code and checks by 2 times. But I think mentors will be able to explain this question better.

Reply:
It's for keeping consistency of test. It's make it more safe for use in that edge-cases, but yeah, wait for mentor

Reply:
> In general, I understand and agree with the logic of the tests, but I have one question: is it necessary to add as many as 3 users to the database for clarity? I would use one, and thus reduce the amount of code and checks by 2 times. But I think mentors will be able to explain this question better.

It's for keeping consistency of test. It's make it more safe for use in that edge-cases, but yeah, wait for mentor